fixtures/alpine/README.md

```diff
@@ -0,0 +1,37 @@
+# `alpine-webserver:v1` Build Process
+
+## Source
+
+The `alpine-webserver:v1` image is based on `alpine` `minirootfs`, but instead of relying on Docker Hub's official Alpine image, we source the original files directly from Alpine's CDN. This approach mitigates issues with Docker Hub and GitHub Action network failures, which have been a common cause of integration test failures.
+
+The build process is specified in the `.github/workflows/kube-integration-tests-non-root.yaml` file.
+
+To avoid distributing Alpine Linux within the Teleport repository, we download it fresh from the Alpine CDN with each build. The download includes the minirootfs (3.3MB), along with the corresponding SHA-256 checksum and GPG signature files for verification.
```

Comment:
> To avoid distributing Alpine Linux within the Teleport repository, we download it fresh from the Alpine CDN with each build

Doesn't that defeat the purpose of the PR?

Reply:
Yes, but given the distribution concerns that @fheinecke mentioned and the greater reliability of the Alpine CDN compared to Docker Hub, I hope we can proceed with this approach.

Ideally, I would prefer to keep the Alpine image within the GitHub repository itself, though this could potentially raise some legal concerns. We can always move to any private repo at any time

Reply:
+1 for Roman's suggestions.

Reply:
Or just manually push it to private ECR and update the workflow to try ECR first and fall back to Docker Hub?

This repo is super bloated - it would be nice to avoid committing additional large binary blobs.

Reply:
@r0mant https://github.com/gravitational/teleport/pull/48321#discussion_r1826310685

I have no idea if the concerns are valid or not. 

@zmb3 Teleport OSS doesn’t have access permissions for our private ECR. We’re using GitHub’s container registry instead to store the buildbox images. We can do the same here but it's probably worth having someone else working on it.

Reply:
> Doesn't that defeat the purpose of the PR?

Downloading before and loading manually in kind will very likely solve the hole issue, the issue can be caused by two things:
- docker hub rate-limits. Since 2022, docker hub enforces agressive pull limits per IP address on unauthenticated requests. Downloading artifacts from a more generous CDN will very likely solve the issue.
- docker in docker networking. Docker in docker setups typically have broken networks caused by MTU mismatch and several NAT layers blocking control frames top reach the caller back. This is a typical dind problem, the dind container has a 1500 MTU while the underlying runnner network has 1460 because of the CNI encapsulation. Downloading the external assets outside of the dind will very likely solve the whole issue and we'll go back to 100% success rate.

Also I agree with @zmb3  and we should try to avoid committing more bloat to the repo.

Reply:
> What are the distribution concerns? It is not a part of the product, we are not modifying it in any way and are not going to distribute it with our release assets. 

Just because it's not a part of the product does not mean that we're not distributing it. This is a ton of software that we don't hold the copyright to, in a publicly-accessible (and indexed) file sharing service. As I see it, the only difference between uploading the image here, and uploading something like Microsoft Windows, Adobe Photoshop, movies, songs, etc. is that we _might_ have a legal right to do so, due to the licenses that each piece of software grants us. There are likely other stipulations to these licenses as well, such as making the source code available as easily as the binaries themselves.

Whether or not we can or should legally do this is a determination that somebody trained in copyright law should make.